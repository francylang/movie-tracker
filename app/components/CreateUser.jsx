import React, { Component } from 'react';
import { Redirect } from 'react-router';

class CreateUser extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      email: '',
      password: ''
    };
  }

  handleChange(event, type) {
    let value = event.target.value;
    if (type === 'email') {
      value = value.toLowerCase();
    }
    this.setState({
      [type]: value
    });
  }


  render() {
    if (this.props.userStatus === true) {
      return <Redirect to='/'/>;
    }

    return (
      <div>
        <form>
          <h2>Welcome to Movie Tracker</h2>
          <h2>Create Account</h2>
          <input
            type='text'
            placeholder='Name'
            onChange={ (event) => this.handleChange(event, 'name' ) }
          />
          <input
            type='text'
            placeholder='Email'
            onChange={ (event) => this.handleChange(event, 'email' ) }
          />
          <input
            type='password'
            placeholder='Password'
            onChange={ (event) => this.handleChange(event, 'password' ) }
          />
          <input
            className='form-button'
            type='submit'
            onClick={
              (event) => {
                event.preventDefault();
                //pass state to action that makes fetch post request to get user back
                const userObj = Object.assign({}, this.state);
                this.props.createNewUser(userObj);
              }
            }
          />
        </form>
      </div>
    );
  }
}

export default CreateUser;
