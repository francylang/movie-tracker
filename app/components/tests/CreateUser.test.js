import React from "react";
import CreateUser from '../CreateUser';
import { shallow } from "enzyme";

describe('CreateUser page', () => {
  const mockFn = jest.fn();
  const wrapper = shallow(<CreateUser userStatus=''
    createNewUser={mockFn}/>);

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should have an empty default state', () => {
    expect(wrapper.state().name).toEqual('');
    expect(wrapper.state().email).toEqual('');
    expect(wrapper.state().password).toEqual('');
  });

  it('should render four inputs', () => {
    const inputs = wrapper.find('input');

    expect(inputs.length).toEqual(4);
  });

  it('should change state on input', () => {
    const nameInput = wrapper.find('.name-input');
    const emailInput = wrapper.find('.email-input');
    const passwordInput = wrapper.find('.password-input');

    nameInput.simulate('change', { target: { value: 'Lola' } });
    emailInput.simulate('change', { target: { value: 'lolabrenner@gmail.com' } });
    passwordInput.simulate('change', { target: { value: 'complete' } });
    expect(wrapper.state().name).toEqual('Lola');
    expect(wrapper.state().email).toEqual('lolabrenner@gmail.com');
    expect(wrapper.state().password).toEqual('complete');
  });

  it('should run function on submit', () => {
    const button = wrapper.find('input').last();

    button.simulate('click', { preventDefault() {} });
    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
