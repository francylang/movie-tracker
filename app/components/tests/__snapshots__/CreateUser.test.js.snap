// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUser page should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateUser
    createNewUser={[Function]}
    userStatus=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <h2>
          Welcome to Movie Tracker
        </h2>
        <h2>
          Create Account
        </h2>
        <input
          className="name-input"
          onChange={[Function]}
          placeholder="Name"
          type="text"
        />
        <input
          className="email-input"
          onChange={[Function]}
          placeholder="Email"
          type="text"
        />
        <input
          className="password-input"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
        <input
          className="form-button"
          onClick={[Function]}
          type="submit"
        />
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Welcome to Movie Tracker
          </h2>,
          <h2>
            Create Account
          </h2>,
          <input
            className="name-input"
            onChange={[Function]}
            placeholder="Name"
            type="text"
          />,
          <input
            className="email-input"
            onChange={[Function]}
            placeholder="Email"
            type="text"
          />,
          <input
            className="password-input"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />,
          <input
            className="form-button"
            onClick={[Function]}
            type="submit"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to Movie Tracker",
          },
          "ref": null,
          "rendered": "Welcome to Movie Tracker",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Create Account",
          },
          "ref": null,
          "rendered": "Create Account",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "name-input",
            "onChange": [Function],
            "placeholder": "Name",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "email-input",
            "onChange": [Function],
            "placeholder": "Email",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "password-input",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "form-button",
            "onClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <h2>
            Welcome to Movie Tracker
          </h2>
          <h2>
            Create Account
          </h2>
          <input
            className="name-input"
            onChange={[Function]}
            placeholder="Name"
            type="text"
          />
          <input
            className="email-input"
            onChange={[Function]}
            placeholder="Email"
            type="text"
          />
          <input
            className="password-input"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <input
            className="form-button"
            onClick={[Function]}
            type="submit"
          />
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Welcome to Movie Tracker
            </h2>,
            <h2>
              Create Account
            </h2>,
            <input
              className="name-input"
              onChange={[Function]}
              placeholder="Name"
              type="text"
            />,
            <input
              className="email-input"
              onChange={[Function]}
              placeholder="Email"
              type="text"
            />,
            <input
              className="password-input"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />,
            <input
              className="form-button"
              onClick={[Function]}
              type="submit"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Movie Tracker",
            },
            "ref": null,
            "rendered": "Welcome to Movie Tracker",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Create Account",
            },
            "ref": null,
            "rendered": "Create Account",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "className": "name-input",
              "onChange": [Function],
              "placeholder": "Name",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "className": "email-input",
              "onChange": [Function],
              "placeholder": "Email",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "className": "password-input",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "className": "form-button",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
